ppl/zqd/handlers.go

```diff
@@ -94,40 +93,26 @@ func handleSearch(c *Core, w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	s, err := c.spaces.Get(req.Space)
-	if err != nil {
-		respondError(c, w, r, err)
-		return
-	}
-
-	ctx, cancel, err := s.StartOp(r.Context())
```

Comment:
I am likely missing, but it appears the StartOp guards are going away in this pr. What happens if a long running search is issued, a delete is issueed which removes directory structure that the search is currently running on? Wouldn't the search get a strange error back?

This problem will be more difficult with zqd as a service, but nevertheless still exists.

Reply:
It is possible that we'll see some "file not found" errors with this change if a delete happens concurrently with the search. I saw an example of that error propagated to Brim, and I also saw an instance where the Brim tab (where the search was running) smoothly stopped the search and flipped to another remaining space.

For the service, I think our flow for this should be that the delete request marks the space id as deleted, but doesn't directly execute the storage removal operation, since that could take an exceedingly long time. The actual removal of data would happen as a background process, scheduled sometime after the delete request, which would have the side effect of allowiong shorter queries/searches to continue unabated.

Reply:
> It is possible that we'll see some "file not found" errors with this change if a delete happens concurrently with the search. I saw an example of that error propagated to Brim, and I also saw an instance where the Brim tab (where the search was running) smoothly stopped the search and flipped to another remaining space.

Seems like we should keep this functionality then, as this leads to unpredictable/inaccurate errors.

Reply:
I retested, and don't think there's a big enough change to hold this back: if we cancel during import with the current Brim release, Brim can show an error that is effectively "path not found" in the current UI. To see the same during a search query, I have to first have a big enough data set that the search lasts long enough for me start the space delete & confirm it. 

I'll file an issue that tracks handling space deletion in a clustered environment, so we can discuss how that should work.